CodePush.m

```diff
@@ -177,6 +179,18 @@ - (void)startRollbackTimer:(int)rollbackTimeout
     resolve(@(isFailedHash));
 }
 
+RCT_EXPORT_METHOD(isFirstRun:(NSString *)packageHash
+                  resolve:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject)
+{
+    NSError *error;
+    BOOL isFirstRun = (nil != packageHash
+                  && [packageHash length] > 0
+                  && [packageHash isEqualToString:[CodePushPackage getCurrentPackageHash:&error]]
+                  && didUpdate);
```

Comment:
Make "didUpdate" the first expression to be evaluated.  It's a simple Boolean check that could make getting the current package hash unnecessary.

Remove the surrounding parentheses.